### PR TITLE
Translate okteto registry at Dockerfiles

### PR DESCRIPTION
**Pull request by teresaromero**
## Proposed changes

- Expand registry when okteto.dev or okteto.global are in the tags of the Dockerfile base images

### CODE REVIEW

File: pkg/cmd/build/build.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ func Run(ctx context.Context, namespace, buildKitHost string, isOktetoCluster bo
 		return err
 	}
 
-	if buildKitHost == okteto.CloudBuildKitURL && dockerFile != "" {
-		dockerFile, err = registry.GetDockerfile(path, dockerFile)
+	if dockerFile != "" {
+		dockerFile, err = registry.GetDockerfile(path, dockerFile, ctx)
```
Comment (pchico83) [inline pkg/cmd/build/build.go:39]: the convention is to use `ctx` as the first argument of any function

File: pkg/registry/file.go [hunk 2]
```diff
@@ -67,6 +68,7 @@ func getDockerfileWithCacheHandler(filename string) (string, error) {
 	for scanner.Scan() {
 		line := scanner.Text()
 		translatedLine := translateCacheHandler(line, userID)
```
Comment (pchico83) [inline pkg/registry/file.go:70]: this should only be done for Okteto Cloud, I think you missed that check

File: pkg/registry/file.go [hunk 3]
```diff
@@ -103,6 +105,32 @@ func translateCacheHandler(input, userID string) string {
 	return input
 }
 
+func translateOktetoRegistry(input string, ctx context.Context) string {
```
Comment (pchico83) [inline pkg/registry/file.go:108]: I think for this case, we can just replace strings.
something like `okteto.dev` -> `registryurl/namespace` and `okteto.global` -> `registryurl/okteto` is a single

Comment (pchico83) [reply pkg/registry/file.go:108]: Also, could you add unit tests for this function?

File: pkg/registry/file.go [hunk 4]
```diff
@@ -103,6 +109,20 @@ func translateCacheHandler(input, userID string) string {
 	return input
 }
 
+func translateOktetoRegistryImage(input, oktetoRegistryURL, namespace string) string {
+
+	if strings.Contains(input, okteto.DevRegistry) {
+		return replaceRegistry(input, okteto.DevRegistry, oktetoRegistryURL, namespace)
```
Comment (pchico83) [inline pkg/registry/file.go:115]: edge case, but you might also need to replace `okteto.GlobalRegistry`.

**Review by jLopezbarb — APPROVED** [review]

File: pkg/okteto/utils.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package okteto
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+// SetToken is used in test to setup token params, returns tempDir and error
+func SetToken(t *Token) (string, error) {
```
Comment (ifbyol) [inline pkg/okteto/utils.go:9]: If this is only used by tests I would have it in an internal package: https://golang.org/doc/go1.4#internalpackages.

I think having the function in a file like `pkg/internal/testutils/utils.go` would let you use the `SetToken` function in the places where it is being used and let others know that it is an utility only for testing purposes

**Review by ifbyol — APPROVED** [review]

File: pkg/okteto/utils.go (hunk 5 above)
Comment (teresaromero) [reply pkg/okteto/utils.go]: I've removed this, as with the new Context is no longer needed :)

**Review by pchico83 — APPROVED** [review]

**Review by ifbyol — APPROVED** [review]

**Pull request merged**